pkg/validation/options.go

```diff
@@ -369,9 +382,10 @@ func parseJwtIssuers(issuers []string, msgs []string) ([]jwtIssuer, []string) {
 
 // newVerifierFromJwtIssuer takes in issuer information in jwtIssuer info and returns
 // a verifier for that issuer.
-func newVerifierFromJwtIssuer(jwtIssuer jwtIssuer) (*oidc.IDTokenVerifier, error) {
+func newVerifierFromJwtIssuer(o *options.Options, jwtIssuer jwtIssuer) (*internaloidc.IDTokenVerifier, error) {
```

Comment:
The jwtIssuer does have an Audience, but this is not the only option required. ClientID and ExtraAudiences are also required to build the IDTokenVerificationOptions struct:

https://github.com/AOEpeople/oauth2-proxy/blob/af820cb0a1818a5b1e7f736ac3654ea5a69a4846/pkg/validation/options.go#L405

Reply:
I'd rather we pass these in directly instead of passing the whole options struct, it ties this function too much to the larger configuration struct right now I think, what about something like this as the signature?
```suggestion
func newVerifierFromJwtIssuer(clientID string, extraAudiences []string, jwtIssuer jwtIssuer) (*internaloidc.IDTokenVerifier, error) {
```

Reply:
This comment hasn't been resolved, WDYT?

Reply:
sorry, I overlooked that. just updated the branch.